decode.go

```diff
@@ -1266,6 +1266,18 @@ func unquoteBytes(s []byte) (t []byte, ok bool) {
 				b[w] = '\t'
 				r++
 				w++
+			case 'x':
+				if (r+2) < len(s) {
+					hex := "0x" + string(s[r+1:r+3])
+					if h, ok := hexString(hex); ok {
```

Comment:
What is this check for? Why not just `strconv.ParseUint(s[r+1:r+3], 16, 8)`?

Reply:
`hexString` checks for correct digits, but since `Parse*int` seems to do the same...

> The errors that ParseInt returns have concrete type *NumError and include err.Num = s. If s is empty or contains invalid digits, err.Err = ErrSyntax and the returned value is 0

...the usage of `hexString` is superfluous.
I will correct it ti simplify it even more.